pythonPackages.faker: fix build

mmlb opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
pythonPackages.faker no longer builds

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

worldofpeace reviewed (changes requested):
Build has a requirement for `freezegun` and why the build seems to be now failing.

Though this did fix the `rm: missing operand`.

mmlb committed:
pythonPackages.faker: fix build and update test deps

Not sure why, but the `__pycache__` folder no longer exists by the time
`postPatch` runs which now causes the `rm` to error and fail the build.

mmlb commented:
@worldofpeace yep I noticed that after a little bit, I've fixed that locally but have now run into a problem that I don't know how to solve. `Faker` depends on `more-itertools<6.0.0` which I can handle here, but then I run into an issue that `pytest` depends on `more-itertools` too and will cause a clash because its using the packaged 6.0.0. I'm stumped on how to fix.

Pushing what I have now.

risicle commented:
You could try feeding the `checkInputs` an ad-hoc `pytest` which has been overridden with your `more-itertools5`:

`(pytest.override { more-itertools = more-itertools5; })`

and if `pytest` won't accept `more-itertools5` as it's too old you could try your luck with `pytest_3`. If you do this you'll probably need to thread this new `pytest` back in to the `pytestrunner` you use.

worldofpeace commented:
> @worldofpeace yep I noticed that after a little bit, I've fixed that locally but have now run into a problem that I don't know how to solve. `Faker` depends on `more-itertools<6.0.0` which I can handle here, but then I run into an issue that `pytest` depends on `more-itertools` too and will cause a clash because its using the packaged 6.0.0. I'm stumped on how to fix.
> 
> Pushing what I have now.

`more-itertools` is just a testing dependency if I'm understanding this correctly.
What kind of failures would result if we used the `6.0.0`, assuming that's why they've enforced the pin.

I'm partial to just patching this away because it's just for tests. Also note we've already patched the pin for `pytest` in `postPatch` so we couldn't do worse.

mmlb commented:
patched to ignore the version specifier (with some notes). @worldofpeace PTAL.

worldofpeace reviewed (approved):
Perfect :+1: 

Builds again for all interpreters.
Please squash your commits and we can merge.

mmlb commented:
done @worldofpeace thanks for help!

worldofpeace commented:
> done @worldofpeace thanks for help!

You're very welcome. Thanks for contributing :sparkles:
